source/fundamentals/auth.txt

```diff
@@ -47,15 +47,18 @@ to instantiate your ``MongoClient``. For descriptions of how to use a
 connection string to specify your authentication method, navigate to the
 relevant mechanism section and select the :guilabel:`Connection String` tab.
 
-The ``MongoCredential`` class contains static factory methods that allow you
-to specify your authentication mechanism and credentials. You can use this
-helper class to keep your code organized. If you choose to use the
-``MongoCredential`` helper class, you need to use the
-``MongoClientSettings.Builder`` class to configure your connection settings
-when constructing your ``MongoClient``. For descriptions of how to use the
-``MongoCredential`` class to specify your authentication method, navigate
-to the relevant mechanism section and select the :guilabel:`MongoCredential`
-tab.
+The ``MongoCredential`` class contains static factory methods that
+construct instances containing your authentication mechanism and
+credentials. Use this helper class to separate your connection details from
+your authentication mechanism and credentials. For example, if you need to
+retrieve your credentials from an external service at runtime, set them in
+a ``MongoCredential`` instead of complicating the connection string
```

Comment:
> instead of complicating the connection string construction

I think you should just remove this phrase.  There is just too much implied in it that won't be obvious.  What you're trying to say is something like this:

1. There is a connection string without credentials stored in a file somewhere, e.g `mongodb://host1,host2/?replicaSet=rs0
2. Credentials are retrieved from some service
3. To construct a new connection string containing these credentials involves splitting the connection string into pieces and creating a new new one with the credential inserted in the right place.
4. It's easier to just use applyConnectionString than to do that

I don't think people will understand that just from this phrase

Reply:
Recap of our zoom conversation:
- Create a ticket for a future task-based learning demonstration of fetching credentials from external service to leverage MongoCredential
- Omit recommendation since it's difficult to convey without showing an example.